migration_steps/prepare_source_data/counts_verification/sql/count_casrec_source.sql

```diff
@@ -0,0 +1,384 @@
+(
+    SELECT COUNT(*) FROM casrec_csv.account
+)
+WHERE supervision_table = 'annual_report_logs';
+
+-- visits
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*) FROM (
+        SELECT DISTINCT vis."casrec_row_id"
+        FROM casrec_csv.REPVIS vis
+--         INNER JOIN countverification.filtered_orders o ON o."Case" = vis."Case"
+    ) t1
+)
+WHERE supervision_table = 'visits';
+
+-- bonds
+UPDATE countverification.counts SET casrec_source =
+(
+    -- active
+    SELECT COUNT(*) FROM casrec_csv.order
+    WHERE casrec_csv.order."Ord Stat" != 'Open'
+    AND casrec_csv.order."Bond Rqd" = 'Y'
+)+(
+    -- dispensed
+    SELECT COUNT(*) FROM casrec_csv.order
+    WHERE casrec_csv.order."Ord Stat" != 'Open'
+    AND casrec_csv.order."Bond Rqd" = 'S'
+)
+WHERE supervision_table = 'bonds';
+
+-- feepayer_id
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*)
+    FROM countverification.filtered_orders o
+    LEFT JOIN casrec_csv.deputyship ds ON ds."Order No" = o."Order No"
+    WHERE ds."Fee Payer" = 'Y'
+)
+WHERE supervision_table = 'feepayer_id';
+
+-- timeline_event
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*) FROM casrec_csv.pat
+)
+WHERE supervision_table = 'timeline_event';
+
+-- person_timeline
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*) FROM casrec_csv.pat
+)
+WHERE supervision_table = 'person_timeline';
+
+-- supervision_level_log
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*)
+    FROM countverification.filtered_orders
+)
+WHERE supervision_table = 'supervision_level_log';
+
+-- finance_invoice_ad
+SET datestyle = "ISO, DMY";
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*) FROM (
+        SELECT DISTINCT "Invoice No"
+        FROM casrec_csv.feeexport fx
+        LEFT JOIN casrec_csv.sop_aged_debt sad ON sad."Trx Number" = fx."Invoice No"
+        WHERE True
+          AND LEFT(fx."Invoice No", 2) = 'AD'
+          AND CAST(fx."Amount" AS DOUBLE PRECISION) > 0
+          AND (NULLIF(fx."Create", 'NaT')::timestamp(0) > '2019-03-31 23:59:59'::timestamp(0)
+                   OR CAST(sad."Outstanding Amount" AS DOUBLE PRECISION) > 0)
+    ) t1
+)
+WHERE supervision_table = 'finance_invoice_ad';
+
+-- finance_invoice_non_ad
+SET datestyle = "ISO, DMY";
+UPDATE countverification.counts SET casrec_source =
+(
+    SELECT COUNT(*) FROM (
+        SELECT DISTINCT "Invoice No"
+        FROM casrec_csv.feeexport fx
+        LEFT JOIN casrec_csv.sop_aged_debt sad ON sad."Trx Number" = fx."Invoice No"
+        WHERE True
+            AND LEFT(fx."Invoice No", 2) <> 'AD'
+            AND CAST(fx."Amount" AS DOUBLE PRECISION) > 0
+            AND (NULLIF(fx."Create", 'NaT')::timestamp(0) > '2019-03-31 23:59:59'::timestamp(0)
+                OR CAST(sad."Outstanding Amount" AS DOUBLE PRECISION) > 0)
+    ) t1
+)
+WHERE supervision_table = 'finance_invoice_non_ad';
```

Comment:
Technically the supervision table is `finance_invoice` and there's no need to separate AD from non-AD for the purpose of count verification... unless you have to map "supervision_table" to a mapping file name, in which case it's fine 😄

Reply:
Cheers Dimitar yes my thinking was more granularity, as with Bonds :)